programs/pluto/ipsec_doi.c

```diff
@@ -463,24 +463,20 @@ ipsecdoi_replace(struct state *st
 		 , lset_t policy_add, lset_t policy_del
 		 , unsigned long try)
 {
-	initiator_function *initiator;
+    initiator_function *initiator;
+    struct connection *c = st->st_connection;
```

Comment:
maybe passert(st!=NULL); first.
Just for sanity.
That destroys the nice initialization, I know.
C99 probably lets us mix declarations and initializations, so I don't mind in 2020, moving forward in C specs.

Reply:
I really don't think a passert() is necessary.  The callers all do lots of manipulation and decisions based on st before calling ipsecdoi_replace()